controllers/workspace/status.go

```diff
@@ -143,6 +145,36 @@ func getIdeUrl(exposedEndpoints map[string]v1alpha1.ExposedEndpointList) string
 	return ""
 }
 
+type PhaseStatus struct {
+	Condition devworkspace.WorkspaceConditionType
+	Message   string
+}
+
+// Defines the order in which DevWorkspace Controller processes phases on reconcile
+// The last unready condition should be shown
+var PhaseStatusesOrder = []PhaseStatus{
+	{
+		devworkspace.WorkspaceComponentsReady,
+		"Processing DevWorkspace components",
+	},
+	{
+		devworkspace.WorkspaceRoutingReady,
+		"Waiting for workspace routing objects",
+	},
+	{
+		devworkspace.WorkspaceServiceAccountReady,
+		"Waiting for workspace serviceaccount",
+	},
+	{
+		PullSecretsReadyCondition,
+		"Waiting for workspace pull secrets",
+	},
+	{
+		devworkspace.WorkspaceReady,
+		"Waiting for deployment to be ready",
+	},
+}
```

Comment:
:+1: good idea!

We should consider (in a separate PR) using `WorkspaceReady` to be the final condition (i.e. after health checks) and add our own `DeploymentReady` condition to use in its place.